ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/PartitionUtils.java

```diff
@@ -94,10 +98,20 @@ public static ExprNodeGenericFuncDesc makeUnaryPredicate(String fn, ExprNodeDesc
   public static Partition getPartition(Hive db, Table table, Map<String, String> partitionSpec, boolean throwException)
       throws SemanticException {
     Partition partition;
-    try {
-      partition = db.getPartition(table, partitionSpec, false);
-    } catch (Exception e) {
-      throw new SemanticException(toMessage(ErrorMsg.INVALID_PARTITION, partitionSpec), e);
+    if (table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned()) {
```

Comment:
that should be done in `Hive.getPartition()`

Reply:
Done. This is not used in this PR.